core/indigo-core/reaction/src/rsmiles_loader.cpp

```diff
@@ -120,15 +119,15 @@ void RSmilesLoader::_loadReaction()
 
     if (_rxn != 0)
     {
-        rcnt.reset(new Molecule());
-        r_loader.loadMolecule((Molecule&)rcnt.ref());
+        rcnt = std::make_unique<Molecule>();
+        r_loader.loadMolecule(static_cast<Molecule&>(*rcnt));
     }
     else
     {
-        rcnt.reset(new QueryMolecule());
-        r_loader.loadQueryMolecule((QueryMolecule&)rcnt.ref());
+        rcnt = std::make_unique<QueryMolecule>();
+        r_loader.loadQueryMolecule(static_cast<QueryMolecule&>(*rcnt));
```

Comment:
I suppose static_cast is appropriate here because the object is created one line earlier and we are absolutely sure about this cast, additional checks are not required. Similarly for other cases